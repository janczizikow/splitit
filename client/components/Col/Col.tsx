import styled from "@emotion/styled";

const Col = styled.div`
  padding: ${p => `0 ${p.theme.gridGutters}px`};
  flex: 1 0 100%;
  max-width: 100%;
  min-height: 1px;
  /* ${p => `${p.theme.gridBreakpoints[1]}px`}
  @media only screen and min-width() {
    flex: 0 0 "x";
    max-width: "x";
  } */
`;

Col.displayName = "Col";

export default Col;
