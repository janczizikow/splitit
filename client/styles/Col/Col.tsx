import styled from "@emotion/styled";
import presets from "../../utils/presets";

interface Props {
  xs?: number;
  sm?: number;
  md?: number;
  lg?: number;
  xl?: number;
  xsOffset?: number;
  smOffset?: number;
  mdOffset?: number;
  lgOffset?: number;
  xlOffset?: number;
  xsOrder?: number;
  smOrder?: number;
  mdOrder?: number;
  lgOrder?: number;
  xlOrder?: number;
}

const Col = styled.div`
  padding: ${p => `0 ${p.theme.gridGutters}px`};
  margin: ${({ xsOffset }) =>
    xsOffset && `0 ${Math.round((xsOffset / 12) * 100)}%`};
  flex: ${(p: Props) => p.xs && `1 0 ${Math.round((p.xs / 12) * 100)}%`};
  max-width: ${(p: Props) => p.xs && `${Math.round((p.xs / 12) * 100)}%`};
  min-height: 1px;
  order: ${({ xsOrder }) => xsOrder};

  ${presets.sm} {
    margin: ${({ smOffset }) =>
      smOffset && `0 ${Math.round((smOffset / 12) * 100)}%`};
    flex: ${({ sm }) => sm && `1 0 ${Math.round((sm / 12) * 100)}%`};
    max-width: ${({ sm }) => sm && `${Math.round((sm / 12) * 100)}%`};
    order: ${({ smOrder }) => smOrder};
  }

  ${presets.md} {
    margin: ${({ mdOffset }) =>
      mdOffset && `0 ${Math.round((mdOffset / 12) * 100)}%`};
    flex: ${({ md }) => md && `1 0 ${Math.round((md / 12) * 100)}%`};
    max-width: ${({ md }) => md && `${Math.round((md / 12) * 100)}%`};
    order: ${({ mdOrder }) => mdOrder};
  }

  ${presets.lg} {
    margin: ${({ lgOffset }) =>
      lgOffset && `0 ${Math.round((lgOffset / 12) * 100)}%`};
    flex: ${({ lg }) => lg && `1 0 ${Math.round((lg / 12) * 100)}%`};
    max-width: ${({ lg }) => lg && `${Math.round((lg / 12) * 100)}%`};
    order: ${({ lgOrder }) => lgOrder};
  }

  ${presets.xl} {
    margin: ${({ xlOffset }) =>
      xlOffset && `0 ${Math.round((xlOffset / 12) * 100)}%`};
    flex: ${({ xl }) => xl && `1 0 ${Math.round((xl / 12) * 100)}%`};
    max-width: ${({ xl }) => xl && `${Math.round((xl / 12) * 100)}%`};
    order: ${({ xlOrder }) => xlOrder};
  }
`;

Col.defaultProps = {
  xs: 12
};

Col.displayName = "Col";

export default Col;
