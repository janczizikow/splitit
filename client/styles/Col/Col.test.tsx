import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Col from "./Col";

describe("<Col />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<Col />);
    expect(wrapper).toHaveLength(1);
  });

  it("matches snapshot", () => {
    const wrapper = mount(<Col />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
