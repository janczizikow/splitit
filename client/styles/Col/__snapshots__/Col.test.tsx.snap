// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col /> matches snapshot 1`] = `
.emotion-0 {
  padding: 0 16px;
  -webkit-flex: 1 0 100%;
  -ms-flex: 1 0 100%;
  flex: 1 0 100%;
  max-width: 100%;
  min-height: 1px;
}

<Col
  theme={
    Object {
      "borderRadius": 2,
      "colors": Object {
        "border": "#bac3c3",
        "error": "#ff483b",
        "errorBg": "#ffebe4",
        "greyBg": "#f9fafa",
        "headingColor": "#2e3333",
        "lightGreyBg": "#f5f5f5",
        "primary": "#00ccbb",
        "success": "#006d68",
        "successBg": "#edfad0",
      },
      "gridBreakpoints": Array [
        0,
        576,
        768,
        992,
        1200,
      ],
      "gridColumns": 12,
      "gridGutters": 16,
    }
  }
  xs={12}
>
  <div
    className="emotion-0"
  />
</Col>
`;
